File: baekjoon_all/24000+/boj_24416.cpp
// Solve 2022-06-03
// Update 2023-10-10

#include <iostream>
using namespace std;

int main() {
    int dp[41] = { 0, 1, 1 };

    for (int i = 3; i < 41; i++) {
        dp[i] = dp[i - 1] + dp[i - 2];
    }

    int n;
    cin >> n;

    cout << dp[n] << ' ' << n - 2 << '\n';

    return 0;
}

File: baekjoon_all/24000+/boj_24416.py
# Solve 2022-06-02
# Update 2023-10-10

n = int(input())

dp = [0] * (n + 1)
dp[1] = dp[2] = 1

for i in range(3, n + 1):
    dp[i] = dp[i - 1] + dp[i - 2]

print(dp[n], n - 2)
